front/src/modules/pathfinding/components/IncompatibleConstraints/type.ts

```diff
@@ -0,0 +1,25 @@
+import type {
+  IncompatibleConstraints,
+  IncompatibleOffsetRange,
+  IncompatibleOffsetRangeWithValue,
+} from 'common/api/osrdEditoastApi';
+
+export type IncompatibleConstraintItemType =
+  | IncompatibleOffsetRange
+  | IncompatibleOffsetRangeWithValue;
+
+export type IncompatibleConstraintTypes = keyof IncompatibleConstraints;
```

Comment:
We try to avoid using `types` in a type name 

```suggestion
export type IncompatibleConstraint = keyof IncompatibleConstraints;
```